core/src/main/scala/org/bitcoins/core/number/NumberType.scala

```diff
@@ -315,13 +315,13 @@ object UInt32
             s"Cannot create ${super.getClass.getSimpleName} from $underlying")
   }
 
-  lazy val zero = UInt32(0)
-  lazy val one = UInt32(1)
+  lazy val zero = UInt32(BigInt(0))
```

Comment:
probably worth leaving a comment that this shouldn't be simplified to avoid infinite loop due to cache code